src/urllib3/util/url.py

```diff
@@ -396,7 +396,7 @@ def parse_url(url):
             path = None
 
     def ensure_type(x):
-        return x if x is None else six.ensure_text(x)
+        return x if x is None else ensure_str(x)
```

Comment:
Idle thought: I feel like this isn't necessary anymore since we only accept URLs that are `str` for Python 3 so URL components can only end up as `str`.